tracer/src/Datadog.Trace/OTelMetrics/MetricReaderHandler.cs

```diff
@@ -0,0 +1,202 @@
+        {
+            ProcessMeasurement(instrument, value, tags, isInteger: false);
+        }
+
+        private static void ProcessMeasurement<T>(Instrument instrument, T value, ReadOnlySpan<KeyValuePair<string, object?>> tags, bool isInteger)
+            where T : struct
+        {
+            try
+            {
+                var instrumentType = instrument.GetType().FullName;
+                if (instrumentType is null)
+                {
+                    Log.Warning("Unable to get the full name of the instrument type for: {InstrumentName}", instrument.Name);
+                    return;
+                }
+
+                ProcessMeasurementCore(instrument, instrumentType, value, tags, isInteger);
+            }
+            catch (Exception ex)
+            {
+                Log.Error(ex, "Error processing measurement for instrument: {InstrumentName}", instrument.Name);
+            }
+        }
+
+        private static void ProcessMeasurementCore<T>(Instrument instrument, string instrumentType, T value, ReadOnlySpan<KeyValuePair<string, object?>> tags, bool isInteger)
+            where T : struct
+        {
+            string aggregationType;
+            if (instrumentType.StartsWith("System.Diagnostics.Metrics.Counter`1") ||
+                instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableCounter`1"))
+            {
+                aggregationType = "Counter";
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.UpDownCounter`1") ||
+                     instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableUpDownCounter`1"))
+            {
+                aggregationType = "UpDownCounter";
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Gauge`1") ||
+                     instrumentType.StartsWith("System.Diagnostics.Metrics.ObservableGauge`1"))
+            {
+                aggregationType = "Gauge";
+            }
+            else if (instrumentType.StartsWith("System.Diagnostics.Metrics.Histogram`1"))
+            {
+                aggregationType = "Histogram";
+            }
+            else
+            {
+                Log.Debug("Skipping unsupported instrument: {InstrumentName} of type: {InstrumentType}", instrument.Name, instrumentType);
+                return;
+            }
+
+            var doubleValue = Convert.ToDouble(value);
+
+            // RFC requirement: Validate negative values for Counter and Histogram
+            if ((aggregationType == "Counter" || aggregationType == "Histogram") && doubleValue < 0)
+            {
+                Log.Warning("Ignoring negative value {Value} for {InstrumentType} instrument: {InstrumentName}. API usage is incorrect.", doubleValue, aggregationType, instrument.Name);
+                return;
+            }
+
+            var meterName = instrument.Meter.Name;
+            var key = string.Concat(meterName, ".", instrument.Name);
```

Comment:
We need to make sure that we have unique metric points based on unique values for all of the following:
- Instrument name
- Instrument kind
- Instrument unit
- Instrument description

Can you make sure they are all reflected in the key?